jscomp/others/js_promise2.res

```diff
@@ -0,0 +1,18 @@
+include Js_promise
+
+/** Type-safe t-first then */
+let then: (promise<'a>, 'a => promise<'b>) => promise<'b> = %raw(`
+  function(p, cont) {
+    Promise.resolve(p).then(cont)
+  }
+  `)
+
+/** Type-safe t-first catch */
+let catch: (promise<'a>, error => promise<'a>) => promise<'a> = %raw(`
+    function(p, cont) {
+      Promise.resolve(p).catch(cont)
+    }
+    `)
+
+/** Old syntax, can use then instead */
+let then_ = then
```

Comment:
Would prefer to not have this in here at all and just copy everything that's really need from Js_promise instead of including Js_promise.

However, if we are going to keep it this way, let's mark `then_` as deprecated (use "then" instead).

Reply:
Deprecated sound good. Will do the self-contained file separately,  as it blurs the changes. (Will squash merge this).